### PR TITLE
Bugfix: Avoid clobbering `handlers.json` before JSON document symbol provider is ready.

### PR DESCRIPTION
**Pull request by mpiroc**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Infrastructure (change which improves the lifecycle management (CI/CD, build, package, deploy, lint, etc) of the application, but does not change functionality.)
- [ ] Technical debt (change which improves the maintainability of the codebase, but does not change functionality)
- [ ] Testing (change which modifies or adds test coverage, but does not change functionality)
- [ ] Documentation (change which modifies or adds documentation, but does not change functionality)

## Description

When symbols are not available for `.aws/handlers.json`, retry with a delay. This handles the case in which the document symbol provider has not yet finished loading during the first attempt.

## Motivation and Context

If the user invokes the 'Configure' code lens, and no JSON documents have previously been opened since VS Code was started, there is a brief window where symbols are not available for that document.

If symbols are unavailable, we consider the document to be invalid (the parser we use, `jsonc-parser`, is highly fault tolerant, so the lack of any symbols indicates a hopelessly malformed document) and create its contents from scratch. But if the symbols are missing due to the document symbol provider still loading, we will unnecessarily clobber the file.

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/lambda/local/configureLocalLambda.ts [hunk 1]
```diff
@@ -126,16 +127,23 @@ function getTabSize(editor?: vscode.TextEditor): number {
...
+        uri
+    ) || await sleep(retryDelay).then(async () => await loadSymbols(uri, maxAttempts - 1, retryDelay))
 }
 
 async function prepareConfig(editor: vscode.TextEditor, handler: string): Promise<boolean> {
-    const symbols: vscode.DocumentSymbol[] | undefined = await getSymbols(editor.document.uri)
+    const symbols: vscode.DocumentSymbol[] | undefined = await loadSymbols(editor.document.uri)
 
     // If the file is empty, or if it is non-empty but cannot be even partially parsed as JSON, build it from scratch.
     if (!symbols || symbols.length < 1) {
```
Comment (awschristou) [inline src/lambda/local/configureLocalLambda.ts]: As an extra line of defense, what do you think about an additional check here, if `symbols` is undefined, to see if the document contains non-whitespace? We could say to the user "hey, looks like there is a file, is it okay if we overwrite it?" <yes> <no (open file)> allowing them to back out. In this way:
* if the json was corrupt, they have a chance to inspect
* if there was json at all, the symbols should be loading for sure by now
* no files are harmed as best we can tell

File: src/lambda/local/configureLocalLambda.ts [hunk 2]
```diff
@@ -126,16 +127,23 @@ function getTabSize(editor?: vscode.TextEditor): number {
...
+): Promise<vscode.DocumentSymbol[] | undefined> {
+    if (maxAttempts <= 0) {
+        return undefined
+    }
 
-    return await vscode.commands.executeCommand<vscode.DocumentSymbol[]>('vscode.executeDocumentSymbolProvider', uri)
+    return await vscode.commands.executeCommand<vscode.DocumentSymbol[]>(
+        'vscode.executeDocumentSymbolProvider',
+        uri
+    ) || await sleep(retryDelay).then(async () => await loadSymbols(uri, maxAttempts - 1, retryDelay))
```
[the remark this answers is not in the document]
Comment (mpiroc) [reply src/lambda/local/configureLocalLambda.ts]: Yeah, I probably got a little too cute here. Will unravel.

File: src/lambda/local/configureLocalLambda.ts (hunk 1 above)
Comment (mpiroc) [reply src/lambda/local/configureLocalLambda.ts]: I agree with this suggestion. I will address it in a separate PR.

File: src/lambda/local/configureLocalLambda.ts [hunk 3]
```diff
@@ -130,16 +130,23 @@ function getTabSize(editor?: vscode.TextEditor): number {
 async function loadSymbols(
     uri: vscode.Uri,
     maxAttempts = 3,
-    retryDelay = 500 // milliseconds
+    retryDelayMillis = 500 // milliseconds
```
Comment (awschristou) [inline src/lambda/local/configureLocalLambda.ts]: Minor: Comment no longer required

**Pull request merged**